src/object/schema_manager.c

```diff
@@ -15106,7 +15106,19 @@ sm_truncate_using_delete (MOP class_mop)
 
   /* We will run a DELETE statement with triggers disabled. */
   save_tr_state = tr_set_execution_state (false);
-  (void) snprintf (delete_query, sizeof (delete_query), "DELETE /*+ RECOMPILE */ FROM %s;", class_name);
+
+  if (pt_is_reserved_word (class_name))
```

Comment:
I don't like to separate the logic. Please just wrap the class name with brackets.